### ISSUE TITLE
[Ansible/artifactory] Using the artifactory_systemyaml_override variable makes role not idempotent

### ISSUE THREAD

**Issue by anderpups** [jfrog/JFrog-Cloud-Installers#203]
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Yes

**Which installer**: Ansible

**Which product and version** (eg: ansible & collection version - 7.24.2): Ansible 2.10, Jfrog Platform Collection 10.3.1

**Which operating system and version**(eg: ubuntu & version - 20.4): Oracle Linux 8

**What happened**: When overriding the default system.yml for Artifactory, the role is no longer idempotent. The creation of the template file is the task that keeps changing even though the file itself doesn't change.

**What you expected to happen**: The second run of the play should have 0 changes

**How to reproduce it** (as minimally and precisely as possible): Run the role twice with the artifactory_systemyaml_override variable set to true. You don't need to provide a artifactory_systemyaml variable since it still does it with the default from the role.

**Anything else we need to know**: I think this may actually be an Ansible bug, but figured I would try here first.

Thanks!

**Comment by anderpups** [comment]
After more troubleshooting it looks like this is due to artifactory encrypting the joinKey and password values in the system.yaml
```
--- before: /opt/jfrog/artifactory/var/etc/system.yaml
+++ after: /home/odtebja/.ansible/tmp/ansible-local-934988i361480w/tmper6kygi9/system.yaml.j2
@@ -1,7 +1,7 @@
 configVersion: 1
 shared:
     security:
-        joinKey: "dea467.aesgcm256.zAJmY_KII5tp-8pFXJe7DJE7_KY6gM3GDV_ef-AAGIm-AfZlxjGP5RraGCnDpJurrLpmfUQT6u_r7M0WEa4Zgs3fYEhgvYxx8spivhBe24p8Gdtq6WyQkPskRW0"
+        joinKey: "[non-encrypted value]"
     extraJavaOpts: "-server -Xms512m -Xmx2g -Xss256k -XX:+UseG1GC"
     node:
         id: "artifactory-app-server"
@@ -13,7 +13,7 @@
         driver: "org.postgresql.Driver"
         url: "jdbc:postgresql://192.168.121.227:5432/artifactory"
         username: "artifactory"
-        password: "dea467.aesgcm256.AubpGA0Hpadpk7Th11J4gzGsRc5tOjoojs6MeCvG3KyXVXocjrR1"
+        password: "[non-encrypted value]"

**Comment by chukka** [comment]
yes, you are right , setting `artifactory_systemyaml_override` makes this not idempotent  because of router in artifactory encrypts password in system.yaml file .

I don't think there is no way to workaround this . please suggest if any workarounds .

**Comment by chukka** [comment]
@anderpups  Do you have any better suggestion to fix this  or can we close this ?

**Comment by anderpups** [comment]
@chukka - Yes please close. I haven't found a good solution either. Thanks
